src/main/java/nl/bioinf/minorapplicationdesign/ontpillen/model/web_scraping/DrugFetcher.java

```diff
@@ -54,38 +59,46 @@ private void storeDrugsInDao(List<Element> drugGroups){
         }
 
         currentDrug = (DrugGroup) drugDao.getDrugByName(currentDrugElement.text());
+        LOGGER.debug("Storing: " + currentDrug.getName() + " in the dao");
 
         if (currentDrugElement.nextElementSibling().is("ul")) {
             List<String> childrenNames = currentDrugElement.nextElementSibling().select("li").eachText();
 
-            for (String childName:childrenNames) {
-                DrugSubstance newDrugSubstance = new DrugSubstance(childName);
-                newDrugSubstance.setParent(currentDrug);
-                currentDrug.addChild(newDrugSubstance);
-                drugDao.addDrug(newDrugSubstance);
-            }
+            addDrugSubstance(currentDrug, childrenNames);
 
         } else {
             String query = currentDrugElement.nextElementSibling().tagName();
             Elements nextGroupSiblings = currentDrugElement.nextElementSiblings().select(query);
 
             List<String> childrenNames = nextGroupSiblings.eachText();
 
-            for (String childName:childrenNames) {
-                DrugGroup newDrugGroup = new DrugGroup(childName);
-                newDrugGroup.setParent(currentDrug);
-                currentDrug.addChild(newDrugGroup);
-                drugDao.addDrug(newDrugGroup);
-            }
+            addDrugGroup(currentDrug, childrenNames);
 
             drugGroups.addAll(nextGroupSiblings);
         }
-
         drugGroups.remove(0);
 
 //        As long as there are elements in this.drugGroups run this method again
         if (drugGroups.size() != 0) {
             storeDrugsInDao(drugGroups);
         }
     }
+
+    private void addDrugGroup(DrugGroup currentDrug, List<String> childrenNames) {
+        for (String childName: childrenNames) {
+            DrugGroup newDrugGroup = new DrugGroup(childName);
+            newDrugGroup.setParent(currentDrug);
+            currentDrug.addChild(newDrugGroup);
+            drugDao.addDrug(newDrugGroup);
+        }
+    }
+
+    private void addDrugSubstance(DrugGroup currentDrug, List<String> childrenNames) {
```

Comment:
De naam even aanpassen naar addDrugSubstances want er worden meteen meerdere genomen, daarnaast currentDrug is logischer hier parentDrug te noemen en childrenNames drugSubstances ofzo